src/main/java/com/conveyal/r5/streets/Split.java

```diff
@@ -98,16 +98,31 @@ public static Split find (double lat, double lon, double searchRadiusMeters,
 
             curr.edge = e;
             edge.seek(e);
+            //LOG.info("Considering OSM way {}", edge.getOSMID());
             //Skip Link edges those are links between transit stops/P+R/Bike share vertices and graph
             //Without this origin or destination point can link to those edges because they have ALL permissions
             //and route is never found since point is inaccessible because edges leading to it don't have required permission
             if (edge.getFlag(EdgeStore.EdgeFlag.LINK)) return true;
 
-            // FIXME this is only checking the forward edge permissions, even though we're iterating over and linking to edge _pairs_
-            // If an edge does not allow traversal with the specified mode, skip over it.
-            if (streetMode == StreetMode.WALK && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_PEDESTRIAN)) return true;
-            if (streetMode == StreetMode.BICYCLE && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_BIKE)) return true;
-            if (streetMode == StreetMode.CAR && !edge.getFlag(EdgeStore.EdgeFlag.ALLOWS_CAR)) return true;
+            if (!edge.allowsStreetMode(streetMode)) {
+                //LOG.info("rejecting edge {} (OSM way {} forward)", edge.edgeIndex, edge.getOSMID());
+                return true;
+            }
+            edge.advance();
+            if (!edge.allowsStreetMode(streetMode)) {
+                //LOG.info("rejecting edge {} (OSM way {} backward)", edge.edgeIndex, edge.getOSMID());
+                return true;
+            }
+            edge.retreat();
+
+//            if (!edge.allowsStreetMode(streetMode)) {
```

Comment:
Is this commented out code duplicating the above?

Reply:
It's not exactly duplicating the logic, the commented out section is providing the other alternative implementation, which accepts an edge if it is car/bike traversable in either direction. The "real" solution is a more complicated set of heuristics to link to multiple roads in the event that they are one-way or dual-carriageway.